addons/editor-theme3/theme3.js

```diff
@@ -1,398 +1,282 @@
-  if (textMode === "colorOnBlack") {
-    stylesheet += `
-      .blocklyDropDownDiv:not([style*="rgb(255, 255, 255)"]) .goog-option-selected .goog-menuitem-checkbox {
-        filter: brightness(0) invert(1);
-      }
-      .u-dropdown-searchbar {
-        border-color: rgba(255, 255, 255, 0.15);
-      }`;
-  }
-  var categories = {
-    motion: {
-      color: "#4C97FF",
-      tertiaryColor: "#3373CC",
-    },
-    looks: {
-      color: "#9966FF",
-      tertiaryColor: "#774DCB",
-    },
-    sounds: {
-      color: "#CF63CF",
-      tertiaryColor: "#BD42BD",
-      alt: "sound",
-    },
-    events: {
-      color: "#DE9E2E",
-      tertiaryColor: "#CC9900",
-    },
-    control: {
-      color: "#FFBF00",
-      tertiaryColor: "#CF8B17",
-    },
-    sensing: {
-      color: "#5CB1D6",
-      tertiaryColor: "#2E8EB8",
-    },
-    operators: {
-      color: "#59C059",
-      tertiaryColor: "#389438",
-    },
-    data: {
-      color: "#FF8C1A",
-      tertiaryColor: "#DB6E00",
-      alt: "variables",
-    },
-    "data-lists": {
-      color: "#FF661A",
-      tertiaryColor: "#E64D00",
-      alt: "lists",
-      var: "dataLists",
-    },
-    custom: {
-      color: "#FF6680",
-      tertiaryColor: "#FF6355",
-      alt: "myBlocks",
-    },
-    Pen: {
-      // For historical reasons, this is called "Pen".
-      color: "#0FBD8C",
-      tertiaryColor: "#0B8E69",
-      alt: "pen",
-    },
-    sa: {
-      color: "#29beb8",
-      tertiaryColor: "#3aa8a4",
-    },
+const extensionsCategory = {
+  id: null,
+  settingId: "Pen-color",
+  colorId: "pen",
+  devtoolsId: "extension",
+};
+const categories = [
+  {
+    id: "motion",
+    settingId: "motion-color",
+    colorId: "motion",
+    devtoolsId: "motion",
+  },
+  {
+    id: "looks",
+    settingId: "looks-color",
+    colorId: "looks",
+    devtoolsId: "looks",
+  },
+  {
+    id: "sound",
+    settingId: "sounds-color",
+    colorId: "sounds",
+    devtoolsId: "sounds",
+  },
+  {
+    id: "events",
+    settingId: "events-color",
+    colorId: "event",
+    devtoolsId: "events",
+  },
+  {
+    id: "control",
+    settingId: "control-color",
+    colorId: "control",
+    devtoolsId: "control",
+  },
+  {
+    id: "sensing",
+    settingId: "sensing-color",
+    colorId: "sensing",
+    devtoolsId: "sensing",
+  },
+  {
+    id: "operators",
+    settingId: "operators-color",
+    colorId: "operators",
+    devtoolsId: "operators",
+  },
+  {
+    id: "variables",
+    settingId: "data-color",
+    colorId: "data",
+    devtoolsId: "data",
+  },
+  {
+    id: null,
+    settingId: "data-lists-color",
+    colorId: "data_lists",
+    devtoolsId: "data-lists",
+  },
+  {
+    id: "myBlocks",
+    settingId: "custom-color",
+    colorId: "more",
+    devtoolsId: null,
+  },
+  extensionsCategory,
+];
+
+export default async function ({ addon, console }) {
+  const Blockly = await addon.tab.traps.getBlockly();
+
+  let originalColors = JSON.parse(JSON.stringify(Blockly.Colours));
```

Comment:
I'm curious why you need to stringify and parse

Reply:
Without this, changing the colors would also affect `originalColors`.